www/app/app.ts

```diff
@@ -252,12 +255,14 @@ export class App {
       console.error(`No server with id ${serverId}`);
       return this.showLocalizedError();
     }
-    const onceNotRunning = server.checkRunning().then((isRunning) => {
-      return isRunning ? this.disconnectServer(event) : Promise.resolve();
-    });
-    onceNotRunning.then(() => {
-      this.serverRepo.forget(serverId);
-    });
+    server.disconnect().then(
```

Comment:
We should check whether the server is running. `server.disconnect` will throw when disconnecting a server that is not connected, failing to forget it.